### PR TITLE
[Docs] Add how to delete project artifacts

### PR DESCRIPTION
**Pull request by jillnogold**
[ML-2585](https://iguazio.atlassian.net/browse/ML-2585)

[ML-2585]: https://iguazio.atlassian.net/browse/ML-2585?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/store/artifacts.md [hunk 1]
```diff
@@ -130,5 +131,21 @@ Example artifact URLs:
 
 <!-- ## Dataset artifacts moved to data coll and prep, model and plot artifacts to working-with-data-and-model-artifacts -->
 
+## Deleting artifacts
+
+Artifacts are comprised of two parts: an artifact object that points to the artifact data; and the artifact data (files). You can delete artifacts from a specific project 
+and choose what you want to delete. Deleting artifacts is supported for v3IO, Google, Azure, DBFS, Filestore, and S3. 
+
+The options for {py:meth}`~mlrun.projects.MlrunProject.delete_artifact`:
```
Comment (moranbental) [inline docs/store/artifacts.md:143]: Why do we need all of the description here if we simply put a link to it: https://docs.mlrun.org/en/latest/api/mlrun.common/mlrun.common.schemas/mlrun.common.schemas.artifact.html#mlrun.common.schemas.artifact.ArtifactsDeletionStrategies

File: docs/store/artifacts.md [hunk 2]
```diff
@@ -130,5 +131,21 @@ Example artifact URLs:
...
+- metadata-only: Delete only the artifact object. The related artifact data remains.
+- data optional: Delete the artifact object and the data. If data deletion is unsuccessful, deletes only the object.
+- data force: Delete the artifact object and the data. If data deletion is unsuccessful, does not delete the object either.
```
Comment (moranbental) [inline docs/store/artifacts.md]:
```suggestion
- data-optional: Delete the artifact object and the data. If data deletion is unsuccessful, deletes only the object.
- data-force: Delete the artifact object and the data. If data deletion is unsuccessful, does not delete the object either.
```

File: docs/store/artifacts.md [hunk 3]
```diff
@@ -130,5 +131,21 @@ Example artifact URLs:
 
 <!-- ## Dataset artifacts moved to data coll and prep, model and plot artifacts to working-with-data-and-model-artifacts -->
 
+## Deleting artifacts
+
+Artifacts are comprised of two parts: an artifact object that points to the artifact data; and the artifact data (files). You can delete artifacts from a specific project 
+and choose what you want to delete. Deleting artifacts is supported for v3IO, Google, Azure, DBFS, Filestore, and S3.
```
Comment (moranbental) [inline docs/store/artifacts.md]:
```suggestion
and choose what you want to delete. Deleting artifacts data is supported for V3IO, Google, Azure, DBFS, Filestore, and S3. 
```
@assaf758 Do we support deletion from `S3` and `Filestore`?

File: docs/projects/create-project.md [hunk 4]
```diff
@@ -254,4 +256,5 @@ Example:
 
 ## Deleting a project
 
-See {py:class}`~mlrun.db.httpdb.HTTPRunDB.delete_project`.
\ No newline at end of file
+See {py:class}`~mlrun.db.httpdb.HTTPRunDB.delete_project`. Deleting a project also deletes the project's objects (artifact metadata, runs, etc.).
```
Comment (moranbental) [inline docs/projects/create-project.md]: When we delete a project, we do not delete the artifact data

**Review by moranbental — CHANGES_REQUESTED** [review]
We also need to specify that we do not support the deletion of the following kinds of artifacts: `ModelArtifact`, `DirArtifact`, and `DatasetArtifact`

File: docs/projects/create-project.md (hunk 4 above)
Comment (jillnogold) [reply docs/projects/create-project.md]: not the files, as written in the next sentence. But the metadata is deleted. Correct?

Comment (moranbental) [reply docs/projects/create-project.md]: The metadata is only removed when the project is deleted with the `deletion_strategy=cascade` option, which is not the default value
https://docs.mlrun.org/en/stable/api/mlrun.db.html#mlrun.db.httpdb.HTTPRunDB.delete_project

Comment (jillnogold) [reply docs/projects/create-project.md]: true, but if there are resources, you can't use the default option. Therefore you can't delete a project and leave the metadata.

File: docs/projects/create-project.md [hunk 5]
```diff
@@ -10,6 +10,7 @@ A project is a container for all the assets, configuration, and code of a partic
 - [Adding functions, artifacts, workflow, and config](#add-elements)
 - [Pushing the project content into git or an archive](#push)
 - [Get a project from DB or create it](#get-or-create)
+- Deleting a project](#deleting-a-project)
```
Comment (theSaarco) [inline docs/projects/create-project.md]:
```suggestion
- [Deleting a project](#deleting-a-project)
```

File: docs/store/artifacts.md [hunk 6]
```diff
@@ -130,5 +131,21 @@ Example artifact URLs:
...
+- metadata-only: Delete only the artifact object. The related artifact data remains.
+- data optional: Delete the artifact object and the data. If data deletion is unsuccessful, deletes only the object.
+- data force: Delete the artifact object and the data. If data deletion is unsuccessful, does not delete the object either.
+
+For example:
+```
+artifact = project.get_artifact("name")
+project.delete_artifact(artifact, deletion_strategy=mlrun.common.schemas.artifact.ArtifactsDeletionStrategies.data_force, secrets={"secret1": "user-secret"})
+```
+Be sure to include `secrets` if credentials are needed to access the artifact data.
```
Comment (theSaarco) [inline docs/store/artifacts.md]:
```suggestion
Be sure to include `secrets` if additional credentials are needed to access the artifact data beyond those already specified as project secrets. 
```

**Review by theSaarco — COMMENTED** [review]
Some small fixes/comments.

File: docs/store/artifacts.md (hunk 1 above)
Comment (jillnogold) [reply docs/store/artifacts.md:143]: It does not hurt to have it here

**Review by moranbental — APPROVED** [review]
LGTM

File: docs/store/artifacts.md [hunk 7]
```diff
@@ -131,5 +132,23 @@ Example artifact URLs:
 
 <!-- ## Dataset artifacts moved to data coll and prep, model and plot artifacts to working-with-data-and-model-artifacts -->
 
+## Deleting artifacts
+
+
+Artifacts are comprised of two parts: an artifact object that points to the artifact data; and the artifact data (files). You delete artifacts from a specific project 
+and choose what you want to delete. Deleting artifacts is supported for v3IO, Google, Azure, DBFS, Filestore, and S3. 
+You cannot delete artifacts of type: ModelArtifact, DirArtifact, and DatasetArtifact.
```
Comment (moranbental) [inline docs/store/artifacts.md]: We are now enabling to delete DatasetArtifact if it just contains one file.

**Review by moranbental — CHANGES_REQUESTED** [review]

**Comment by jillnogold** [comment]
@moranbental  - updated - pls check again

**Review by moranbental — APPROVED** [review]
LGTM

File: docs/projects/create-project.md [hunk 8]
```diff
@@ -279,4 +282,7 @@ project.run("main", arguments={"data": data_url})  # run the workflow "main"
 
 ## Deleting a project
 
-See {py:class}`~mlrun.db.httpdb.HTTPRunDB.delete_project`.
\ No newline at end of file
+See {py:class}`~mlrun.db.httpdb.HTTPRunDB.delete_project` 
+also deletes the project's objects (artifact metadata, runs, etc.), 
+but not the artifact data. (When deleting a project that has resources, you must use `deletion_strategy=cascade`, which deletes the objects.)
```
Comment (alonmr) [inline docs/projects/create-project.md]:
```suggestion
but not the artifact data. (When deleting a project that has resources, you must use `deletion_strategy=mlrun.common.schemas.DeletionStrategy.cascade`, which deletes the objects.)
```

File: docs/store/artifacts.md [hunk 9]
```diff
@@ -131,5 +132,23 @@ Example artifact URLs:
 
 <!-- ## Dataset artifacts moved to data coll and prep, model and plot artifacts to working-with-data-and-model-artifacts -->
 
+## Deleting artifacts
+
+
+Artifacts are comprised of two parts: an artifact object that points to the artifact data; and the artifact data (files). You delete artifacts from a specific project 
+and choose what you want to delete. Deleting artifacts is supported for v3IO, Google, Azure, DBFS, Filestore, and S3.
```
Comment (alonmr) [inline docs/store/artifacts.md]:
```suggestion
and choose what you want to delete. Deleting artifacts data is supported for V3IO, Google, Azure, DBFS, Filestore, and S3. 
```

**Review by alonmr — APPROVED** [review]

**Pull request merged**